compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2570,9 +2571,25 @@ object Types {
               || adapted.info.eq(denot.info))
             adapted
           else this
-        result.setDenot(denot)
+        val lastDenot = result.lastDenotation
+        denot match
+          case denot: SymDenotation
+          if denot.validFor.firstPhaseId < ctx.phase.id
+            && lastDenot != null
```

Comment:
I'm not completely sure that we won't run into issues where lastDenot didn't exist, but we need to get the denotation at an earlier phase in a subsequent run or via time-travelling. How expensive would it be to always create UniqueRefDenotation after erasure?

Reply:
I prefer caution here. I don't have the time to look into a change like this, which looks risky to me.